doc/source/operations/upgrading.rst

```diff
@@ -0,0 +1,726 @@
+
+Preparation
+===========
+
+Preparation is crucial for a successful upgrade. It allows for a minimal
+maintenance/change window and ensures we are ready if unexpected issues arise.
+
+Upgrade plan
+------------
+
+The less you need to think on upgrade day, the better. Save your brain for
+solving any issues that arise. Write an upgrade plan detailing:
+
+* the predicted schedule
+* a checklist of prerequisites
+* a set of smoke tests to perform after significant changes
+* a list of steps to perform during the preparation phase
+* a list of steps to perform during the upgrade maintenance/change window phase
+* a list of steps to perform during the follow up phase
+* space to make notes of progress and any issues/solutions/workarounds that
+  arise
+
+Ideally all steps will include the exact commands to execute that can be
+copy/pasted, or links to appropriate CI/CD workflows to run.
+
+Backing up
+----------
+
+Before you start, be sure to back up any local changes, configuration, and
+data.
+
+See the :kayobe-doc:`Kayobe documentation
+<administration/overcloud.html#performing-database-backups>` for information on
+backing up the overcloud MariaDB database. It may be prudent to take backups at
+various stages of the upgrade since the database state will change over time.
+
+Updating code forks
+-------------------
+
+If the deployment uses any source code forks (other than the StackHPC ones),
+update them to use the |current_release| release.
+
+Migrating Kayobe Configuration
+------------------------------
+
+Kayobe configuration options may be changed between releases of Kayobe. Ensure
+that all site local configuration is migrated to the target version format.
+See the :skc-doc:`StackHPC Kayobe Configuration release notes
+<release-notes.html>`, :kayobe-renos:`Kayobe release notes <>` and
+:kolla-ansible-renos:`Kolla Ansible release notes <>`. In particular, the
+*Upgrade Notes* and *Deprecation Notes* sections provide information that might
+affect the configuration migration.
+
+In the following example we assume a branch naming scheme of
+``example/<release>``.
+
+Create a branch for the new release::
+
+    git fetch origin
+    git checkout example/<previous release>
+    git checkout -b example/|current_release|
+    git push origin example/|current_release|
+
+Merge in the new branch of StackHPC Kayobe Configuration::
+
+    git remote add stackhpc https://github.com/stackhpc/stackhpc-kayobe-config
+    git fetch stackhpc
+    git fetch origin
+    git checkout -b example/|current_release|-sync origin/example/|current_release|
+    git merge stackhpc/|current_release_git_branch_name|
+
+There may be conflicts to resolve. The configuration should be manually
+inspected after the merge to ensure that it is correct. Once complete, push the
+branch and create a pull request with the changes::
+
+    git push origin example/|current_release|-sync
+
+Once approved and merged, update the configuration to adapt to the new release.
```

Comment:
Might be worth reminding people here that the base environment ``etc/kayobe/foo.yml`` will likely have changes that are brought in from the merge, but anything in a deployment-specific environment will be completely unaffected.

Reply:
Good idea.